Mathlib/Logic/Function/CompTypeclasses.lean

```diff
@@ -0,0 +1,65 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+
+import Mathlib.Init.Function
+
+/-!
+# Propositional typeclasses on several maps
+
+This file contains typeclasses that are used in the definition of
+equivariant maps in the spirit what was initially developed
+by Frédéric Dupuis and Heather Macbeth for linear maps.
+
+* `CompTriple φ ψ χ`, which expresses that `ψ.comp φ = χ`
+* `IsId φ`, which expresses that `φ = id`
+
+TODO :
+* align with RingHomCompTriple
+
+-/
+
+section CompTriple
+
+/-- Class of composing triples -/
+class CompTriple  {M N P : Type*} (φ : M → N) (ψ : N → P) (χ : outParam (M → P)) : Prop where
+  /-- The maps form a commuting triangle -/
+  comp_eq : ψ.comp φ = χ
+
+attribute [simp] CompTriple.comp_eq
+
+namespace CompTriple
+
+/-- Class of Id maps -/
+class IsId {M : Type*} (σ : M → M) : Prop where
+  eq_id : σ = id
+
+instance {M : Type*} : IsId (@id M) where
+  eq_id := rfl
+
+lemma comp_id {N P : Type*} {φ : N → N} [IsId φ] {ψ : N → P} :
+    CompTriple φ ψ ψ where
```

Comment:
As in the other file, it would be good to know why this (and the lemmas below) are not listed as instances.

Reply:
I am trying to, let's see…